### ISSUE TITLE
How to encode a Literal Header Field with Incremental Indexing?

### ISSUE THREAD

**Issue by shansjlin** [nghttp2/nghttp2#530]
When I readed the document [here](http://tools.ietf.org/html/rfc7541#page-15) , I found that there are two different ways to 
encode a Literal Header Field:
1.Literal Header Field with Incremental Indexing
2.Literal Header Field without Indexing

And I found the enum type in nghttp2.h header file as shown below:
`typedef enum {
  NGHTTP2_NV_FLAG_NONE = 0,
  NGHTTP2_NV_FLAG_NO_INDEX = 0x01,
  NGHTTP2_NV_FLAG_NO_COPY_NAME = 0x02,
  NGHTTP2_NV_FLAG_NO_COPY_VALUE = 0x04
} nghttp2_nv_flag;`

I only found the NGHTTP2_NV_FLAG_NO_INDEX  flag, so does it mean that:
I just need to set the flag to be NGHTTP2_NV_FLAG_NONE  
if I want to encode a Literal Header Filed with Incremental Indexing?

And [here](http://tools.ietf.org/html/rfc7541#appendix-A) is the static table for incremental index

**Comment by tatsuhiro-t** [comment]
I guess you are reading Apple's APNs documentation.  If so, forget about HPACK stuff in their documentation.  It just goes too deep down to HTTP/2 HPACK mechanism, which APNs application developer does not have to need to know.  nghttp2 will take care of all HPACK stuff for you.

**Comment by linyanwen** [comment]
Yes, I am reading APNs documentation, and I found a strange question is when I post an APNs Notification request, and Apple`s server return success with the :status=200, but my iphone device doesn`t get the APNs Notification,  so I doubt that if I don`t encode the header fields in the right way.
But as you said, nghttp2 will take care of all HPACK stuff,  I probably made a mistake in other places.
Anyway, Thanks a lot.

**Comment by tatsuhiro-t** [comment]
If server returns HTTP status, it most likely means HPACK encoded header block was correctly decoded on server side.  So probably, you may be missing something in required header fields or wrong device ID (I'm not sure).
